addons/event/models/event.py

```diff
@@ -212,6 +213,8 @@ def create(self, vals):
 
     @api.multi
     def write(self, vals):
+        if vals.get('date_begin') or vals.get('date_end') or vals.get('address_id'):
+            self.rescheduled_alert = True
```

Comment:
do not set self.rescheduled_alert here. set value of rescheduled_alert in vals dict instead  of it.